site/profile/manifests/userportal.pp

```diff
@@ -0,0 +1,121 @@
+class profile::userportal::server (
+  $password
+){
+  $domain_name = lookup('profile::freeipa::base::domain_name')
+  package {['python3-virtualenv', 'python38', 'python38-devel']: }
+  package {['openldap-devel', 'gcc', 'mariadb-devel']: }
+
+  # Using python3.8 with gunicorn
+  exec { 'create virtualenv':
+    command => '/usr/bin/virtualenv-3 --python="/usr/bin/python3.8" /var/www/userportal-env',
+    unless  => '/usr/bin/test -d /var/www/userportal-env',
+    require => [Package['python3-virtualenv'], Package['python38']],
+  }
+
+  file { '/var/www/userportal/':
+    ensure => 'directory',
+    owner  => 'apache',
+    group  => 'apache',
+  }
+  -> vcsrepo { '/var/www/userportal/':
+    ensure   => present,
+    provider => git,
+    source   => 'https://github.com/guilbaults/TrailblazingTurtle.git',
+    revision => '7940ae14891a60d18afd1d9d009dada044512b0f',
+    user     => 'apache',
+    notify   => [Service['httpd'], Service['gunicorn']],
+  }
+  -> file { '/var/www/userportal/userportal/settings.py':
+    show_diff => false,
+    content   => epp('profile/userportal/settings.py',
+      {
+        'password'     => $password,
+        'cluster_name' => lookup('profile::slurm::base::cluster_name'),
+        'secret_key'   => fqdn_rand_string(32, undef, $password),
+        'domain_name'  => $domain_name,
+      }
+    ),
+    notify    => Service['gunicorn'],
+  }
+  -> file { '/var/www/userportal/userportal/common.py':
+    source => 'file:/var/www/userportal/example/common.py',
+    notify => [Service['httpd'], Service['gunicorn']],
```

Comment:
We probably need to only notify gunicorn if common.py is modified/created here no?

Reply:
Only gunicorn need to be notified there, HTTPD needs to be notified a few lines above because the static file path can be changed in the settings.py. (fixed)